src/org/exist/http/RESTServer.java

```diff
@@ -1854,26 +1854,24 @@ private void writeXPathException(final HttpServletResponse response,
 
         response.setContentType(MimeType.XML_TYPE.getName() + "; charset=" + encoding);
 
-        final OutputStreamWriter writer =
-                new OutputStreamWriter(response.getOutputStream(), encoding);
+        try(final OutputStreamWriter writer =
+                new OutputStreamWriter(response.getOutputStream(), encoding)) {
 
-        writer.write("<?xml version=\"1.0\" ?>");
-        writer.write("<exception><path>");
-        writer.write(path);
-        writer.write("</path>");
-        writer.write("<message>");
-        final String message = e.getMessage() == null ? e.toString() : e.getMessage();
-        writer.write(XMLUtil.encodeAttrMarkup(message));
-        writer.write("</message>");
-        if (query != null) {
-            writer.write("<query>");
-            writer.write(XMLUtil.encodeAttrMarkup(query));
-            writer.write("</query>");
+            writer.write("<?xml version=\"1.0\" ?>");
+            writer.write("<exception><path>");
+            writer.write(path);
+            writer.write("</path>");
+            writer.write("<message>");
+            final String message = e.getMessage() == null ? e.toString() : e.getMessage();
+            writer.write(XMLUtil.encodeAttrMarkup(message));
+            writer.write("</message>");
+            if (query != null) {
+                writer.write("<query>");
+                writer.write(XMLUtil.encodeAttrMarkup(query));
+                writer.write("</query>");
+            }
+            writer.write("</exception>");
```

Comment:
Don't we need a flush() call here to be sure all data is sent to the http agent?

Reply:
answering myself: no not according to the servlet spec :-)